packages/evmConnectors/EvmWeb3authConnector/src/EvmWeb3authConnector.ts

```
@@ -0,0 +1,107 @@
+import core, {
+  EvmAddress,
+  MoralisCore,
+  EvmChain,
+  MoralisNetworkConnectorError,
+  EvmWeb3authConnectOptions,
+  NetworkConnectorErrorCode,
+  EvmProvider,
+  EvmConnection,
+} from '@moralisweb3/core';
+import { EvmAbstractConnector } from '@moralisweb3/evm-connector-utils';
+import { Web3Auth, Web3AuthOptions } from '@web3auth/web3auth';
+import { SafeEventEmitterProvider, ADAPTER_EVENTS } from '@web3auth/base';
+
+const DEFAULT_OPTIONS: Omit<EvmWeb3authConnectOptions, 'clientId'> = {
+  theme: 'dark',
+  appLogo: 'https://moralis.io/wp-content/uploads/2021/05/moralisWhiteLogo.svg',
+  chainId: '0x1',
+};
+
+export type Web3AuthProvider = EvmProvider & SafeEventEmitterProvider;
+
+export interface EvmWeb3authConnectorConfig {
+  core: MoralisCore;
+}
+export class EvmWeb3authConnector extends EvmAbstractConnector<Web3AuthProvider, EvmWeb3authConnectOptions> {
+  constructor(config: EvmWeb3authConnectorConfig) {
+    super({
+      name: 'web3auth',
+      core: config.core,
+    });
+  }
+
+  protected async createProvider(params: EvmWeb3authConnectOptions): Promise<Web3AuthProvider> {
+    const options: Web3AuthOptions = {
+      chainConfig: {
+        chainId: EvmChain.create(params.chainId!).apiHex,
+        chainNamespace: 'eip155',
+        blockExplorer: params.blockExplorer,
+        displayName: params.displayName,
+        rpcTarget: params.rpcTarget,
+        ticker: params.ticker,
+        tickerName: params.tickerName,
```

Since these are added, it does not work anymore.
If you don't supply a rpcTarget, then it will throw an error (instead of defaulting)

This approach should work:
```
        ...(params.displayName && { displayName: params.displayName }),
        ...(params.rpcTarget && { rpcTarget: params.rpcTarget }),
        ...(params.ticker && { ticker: params.ticker }),
        ...(params.tickerName && { tickerName: params.tickerName }),
```